scripts/ci/pr-bot/processPrUpdate.ts

```diff
@@ -0,0 +1,198 @@
+    await stateClient.writePrState(pullNumber, prState);
+    await stateClient.writeReviewersForLabelState(
+      labelOfReviewer,
+      reviewersState
+    );
+  }
+}
+
+// On pr push or author comment, we should put the attention set back on the reviewers
+async function setNextActionReviewers(
+  payload,
+  stateClient: typeof PersistentState
+) {
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+  if (!(await areReviewersAssigned(pullNumber, stateClient))) {
+    console.log("No reviewers assigned, dont need to manipulate attention set");
+    return;
+  }
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  await nextActionReviewers(pullNumber, existingLabels);
+  let prState = await stateClient.getPrState(pullNumber);
+  prState.nextAction = "Reviewers";
+  await stateClient.writePrState(pullNumber, prState);
+}
+
+async function processPrUpdate() {
+  const reviewerConfig = new ReviewerConfig(PATH_TO_CONFIG_FILE);
+  const context = github.context;
+  console.log("Event context:");
+  console.log(context);
+  const payload = context.payload;
+
+  // TODO(damccorm) - remove this when we roll out to more than go
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  if (!existingLabels.find((label) => label.name.toLowerCase() == "go")) {
+    console.log("Does not contain the go label - skipping");
+    return;
+  }
+
+  if (!payload.issue?.pull_request && !payload.pull_request) {
+    console.log("Issue, not pull request - returning");
+    return;
+  }
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+
+  let stateClient = new PersistentState();
+  let prState = await stateClient.getPrState(pullNumber);
+  if (prState.stopReviewerNotifications) {
+    console.log("Notifications have been paused for this pull - skipping");
+    return;
+  }
+
+  switch (github.context.eventName) {
+    case "pull_request_review_comment":
+    case "issue_comment":
+      console.log("Processing comment event");
+      if (payload.action != "created") {
+        console.log("Comment wasnt just created, skipping");
+        return;
+      }
+      await processPrComment(payload, stateClient, reviewerConfig);
+      break;
+    case "pull_request_review":
+      console.log("Processing PR review event");
+      await processPrReview(payload, stateClient, reviewerConfig);
+      break;
+    case "pull_request_target":
+      if (payload.action == "synchronize") {
+        console.log("Processing synchronize action");
+        await setNextActionReviewers(payload, stateClient);
+      }
+      // TODO(damccorm) - it would be good to eventually handle the following events here, even though they're not part of the normal workflow
+      // review requested, assigned, label added, label removed
+      break;
+    default:
+      console.log("Not a PR comment, push, or review, doing nothing");
+  }
+}
+
+processPrUpdate();
+
+export {};
```

Comment:
I've never seen this self executing pattern before. I think exporting `processNewPrs` and then calling it in the index.ts file would be more explicit. Hiding execution under an import could make debugging more difficult later.

Reply:
Addressed in a similar comment above